### PR TITLE
Fix to return 401 instead of 500

### LINKED ISSUE #707: 401 instead of 500 for bad credentials
Currently, passing bad credentials is returning 500 instead of 401.

### PR DESCRIPTION
**Pull request by kathirsvn**
**What this PR does**:

Fix to return 401 instead of 500 when invalid/token or credentials are provided

**Which issue(s) this PR fixes**:
Fixes #707 

**Checklist**
- [X] Changes manually tested
- [X] Automated Tests added/updated
- [X] Documentation added/updated
- [X] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java [hunk 1]
```diff
@@ -70,10 +71,40 @@ public final class ThrowableToErrorMapper {
...
+                        .filter(
+                            t ->
+                                t instanceof AuthenticationException
+                                    || t instanceof IllegalArgumentException)
+                        .orElse(null);
+                // connecting to oss cassandra throws AuthenticationException for invalid
+                // credentials connecting to AstraDB throws IllegalArgumentException for invalid
+                // token/credentials
+                if (error instanceof AuthenticationException
+                    || (error instanceof IllegalArgumentException
```
Comment (Yuqi-Du) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:93]: Oh, So IllegalArgumentException is the issue.

**Review by Yuqi-Du — APPROVED** [review]

**Comment by tatu-at-datastax** [comment]
Looks good in general.

One question: is there any way to be able to get `errorCode` to be added in `CommandResult.Error` returned? There are discussion with Client-side folks to be able to rely more on `errorCode` for classifying types of failures.
But as things are, only `JsonApiException` produces `ErrorCode` values.

I guess this is a bigger question and maybe not something to tackle for this PR, but I thought I'd mention it for future reference.

**Review by vkarpov15 — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java (hunk 1 above)
Comment (kathirsvn) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:93]: Yes, correct.

**Comment by kathirsvn** [comment]
> Looks good in general.
> 
> One question: is there any way to be able to get `errorCode` to be added in `CommandResult.Error` returned? There are discussion with Client-side folks to be able to rely more on `errorCode` for classifying types of failures. But as things are, only `JsonApiException` produces `ErrorCode` values.
> 
> I guess this is a bigger question and maybe not something to tackle for this PR, but I thought I'd mention it for future reference.

Agreed. We can discuss further on this in [this new issue](https://github.com/stargate/jsonapi/issues/720).

**Pull request merged**